app/assets/javascripts/puppetclasses_or_envs_changed.js

```diff
@@ -0,0 +1,12 @@
+$(function () {
+
+  var last_target = null;
+
+  $('[data-toggle="popover"]').on('click', function (e) {
+    if ($('.popover:visible').length && last_target != e.target)
+      $(last_target).popover('toggle');
+
+    $(e.target).popover('toggle');
+    last_target = e.target;
```

Comment:
```javascript
 $('[data-toggle="popover"]').on('click', function () {
    var prev_popover = $('.popover:visible').prev('a');
    if (!$(this).is(prev_popover))
      prev_popover.popover('toggle');
    $(this).popover('toggle');
  });
```
Should work just as well and a bit cleaner I think, not relaying on the last_target being correctly saved but rather on the current status of the popovers.